lisa/sut_orchestrator/azure/common.py

```diff
@@ -1904,3 +1914,270 @@ def get_size(disk_type: schema.DiskType, data_disk_iops: int = 1) -> int:
             return iops_dict[min_iops]
         else:
             raise LisaException(f"Data disk type {disk_type} is unsupported.")
+
+
+def add_system_assign_identity(
+    platform: "AzurePlatform",
+    resource_group_name: str,
+    vm_name: str,
+    location: str,
+    log: Logger,
+) -> Any:
+    compute_client = ComputeManagementClient(
+        platform.credential, platform.subscription_id
+    )
+    params_identity = {"type": "SystemAssigned"}
+    params_create = {"location": location, "identity": params_identity}
+
+    vm_poller = compute_client.virtual_machines.begin_create_or_update(
+        resource_group_name,
+        vm_name,
+        params_create,
+    )
+    vm_result = vm_poller.result()
+    object_id_vm = vm_result.identity.principal_id
+    log.info(object_id_vm)
+
+    if not object_id_vm:
+        raise ValueError("object_id_vm is not set.")
```

Comment:
make it more descriptive, like "cannot retrieve managed identity after set system assigned identity on vm."